cirq/ops/moment.py

```diff
@@ -92,7 +92,15 @@ def with_operation(self, operation: raw_types.Operation):
         Returns:
             The new moment.
         """
-        return Moment(self.operations + (operation,))
+        if any(q in self._qubits for q in operation.qubits):
+            raise ValueError('Overlapping operation: {}'.format(operation))
+
+        # Use private variables to facilitate a quick copy
```

Comment:
This is because you are trying to avoid the overlap check being evaluated?

Reply:
I am trying to skip the whole __init__ function, so I have to replicate the overlap check as well.  I am basically re-implementing the __init__ function for this special case (which is used heavily in circuit initialization)